sound/soc/sof/topology.c

```diff
@@ -1134,11 +1152,41 @@ static int sof_widget_parse_tokens(struct snd_soc_component *scomp, struct snd_s
 			}
 
 			continue;
+		case SOF_IPC4_IN_AUDIO_FORMAT_TOKENS:
+		case SOF_IPC4_OUT_AUDIO_FORMAT_TOKENS:
+		case SOF_IPC4_COPIER_GATEWAY_CFG_TOKENS:
+		case SOF_IPC4_AUDIO_FORMAT_BUFFER_SIZE_TOKENS:
```

Comment:
I did not expected to see *IPC4* in generic code.

Reply:
That is a good topic. They are currently all IPC4 specific, but I agree that removing the IPC4 prefix is a good idea.

Reply:
Even if the IPC4 is removed from the defines, this is really IPC4. If the object is either of these four types (TOKENS) then we query the number of AUDIO_FORMATS (read the value of that token).

`sof_get_token_value()` is local to topology.c, so it might be a bit of an issue, but what if we have an ops let's say, `get_num_sets(scomp, swidget, object_token_list[i])` and the IPC dependent code would return with the number of sets for the given token type. If the ops is not implemented then we will have `num_sets = 1`

```c
int num_sets = 1;

if (object_token_list[i] == SOF_COMP_EXT_TOKENS) {
...
}

if (ipc_tplg_ops->get_num_sets) {
	num_sets = ipc_tplg_ops->get_num_sets(scom, swidget, object_token_list[i])
	if (num_sets < 0) {
		dev_err(sdev->dev, "Invalid audio format count for %s\n",
			swidget->widget->name);
		ret = num_sets;
		goto err;
	}
}

if (num_sets > 1) {
// realloc
}
```

this way this can be generic and not tied to ipc4.

Reply:
I will remove the IPC4 prefix first.